### ISSUE TITLE
Vertical pagination

### ISSUE THREAD

**Issue by ritz078** [rogerpasky/staffrender#2]
Currently this library supports horizontal pagination which means:

- I can't use the whole available space on screen.
- It's hard to follow an active note when the page suddenly changes. Vertical pagination is often easier to follow as I can always see what's coming in the next page.

I am using @magenta/music and realised this is being used there. Thanks for the awesome library.

**Comment by rogerpasky** [comment]
Thank you for your comments.

Original motivation was to extend existing Magenta Visualizer, which was just a horizontal-scrolled piano roll.

This is an on-progress library which has vertical scrolling in mind, but there are some other improvements far much needed, like triplets or rhythm-grouped beams (for shorter than quarter notes), which I'm involved with nowadays. Triplets are easy, but properly grouping is a nightmare (official rules, plenty of exceptions, are incompletely covered in more than 20 pages on Berkeley books). I can't promise vertical scroll will arrive this year, maybe next one.

Regarding to the active note following, there are three auto-scrolling modes: BAR, NOTE and PAGE as detailed in [ documentation](https://rogerpasky.github.io/staffrender/enums/_staff_svg_render_.scrolltype.html). Probably you are using PAGE, which is default value, but you will probably enjoy BAR mode, which scrolls once each bar ends, making it easier to follow when playing. NOTE mode may be too staggered to follow fast scores. 

In order to do it so, just set it in the [StaffRenderConfig](https://rogerpasky.github.io/staffrender/interfaces/_staff_svg_render_.staffrenderconfig.html) on the constructor invocation, or similar in the equivalent Config parameter at @magenta/music.

Hope this helps for the waiting.

**Comment by ritz078** [comment]
> This is an on-progress library which has vertical scrolling in mind, but there are some other improvements far much needed, like triplets or rhythm-grouped beams (for shorter than quarter notes), which I'm involved with nowadays. Triplets are easy, but properly grouping is a nightmare (official rules, plenty of exceptions, are incompletely covered in more than 20 pages on Berkeley books). I can't promise vertical scroll will arrive this year, maybe next one.

Totally agree to this. Thank you for putting your time and effort in this amazing library. It important to get the notes right first just like you said. 

> Regarding to the active note following, there are three auto-scrolling modes: BAR, NOTE and PAGE as detailed in documentation. Probably you are using PAGE, which is default value, but you will probably enjoy BAR mode, which scrolls once each bar ends, making it easier to follow when playing. NOTE mode may be too staggered to follow fast scores.

I tried all 3 and realised that no one is smooth on the eyes while you are playing a piano together. I even tried with smooth scroll. Here is how far I was able to reach https://twitter.com/ritz078/status/1300039283168800768. The code is present here: https://github.com/ritz078/raaga/pull/112. I will try to dive into the codebase.

Again thank you for this awesome library.
